lib/Doctrine/Persistence/Mapping/Driver/FileDriver.php

```diff
@@ -34,15 +35,17 @@ abstract class FileDriver implements MappingDriver
      * Initializes a new FileDriver that looks in the given path(s) for mapping
      * documents and operates in the specified operating mode.
      *
-     * @param string|string[]|FileLocator $locator A FileLocator or one/multiple paths
-     *                                          where mapping documents can be found.
+     * @param string|array<int, string>|FileLocator $locator A FileLocator or one/multiple paths
+     *                                                       where mapping documents can be found.
      */
     public function __construct($locator, ?string $fileExtension = null)
     {
         if ($locator instanceof FileLocator) {
             $this->locator = $locator;
         } else {
-            $this->locator = new DefaultFileLocator((array) $locator, $fileExtension);
+            $paths = is_string($locator) ? [$locator] : $locator;
```

Comment:
An `(array)` cast was correct here (assuming `array<string>|string`

Reply:
Do you prefer the the casting over this? I thought this might be more desirable/explicit over the cast.

Reply:
I prefer array cast. Less branches

Reply:
A cast is safe in some scenarios, and it is indeed a more efficient operation.

Casting is often risky, but here we know what is going on, IMO

Reply:
Fixed